bigtable/bigtable.go

```diff
@@ -467,13 +479,30 @@ func RowFilter(f Filter) ReadOption { return rowFilter{f} }
 type rowFilter struct{ f Filter }
 
 func (rf rowFilter) set(req *btpb.ReadRowsRequest) { req.Filter = rf.f.proto() }
+func (rf rowFilter) getRequestStatsCallback() RequestStatsCallback {
```

Comment:
Hmm I'm unfamiliar with go syntax but why do we need to getRequestStatsCallback for rowFilter or Row limit options? And what does getRequestStatsCallback do?

Reply:
Since the `ReadOption` interface includes the new method `getRequestStatsCallback` signature, for `rowFilter` and `limitRows` to adhere to this interface, they need to include an implementation of that method.

Go's interface system is not totally clear to readers IMO, but Go will automatically mark `rowFilter` and `limitRows` as implementing this interface at compile time as long as the methods match up.

Without these changes, `RowFilter` above breaks since it indicates that it returns a `ReadOption`, which means that `rowFilter` must have a method that matches the `getRequestStatsCallback` spec, else it can't be considered as part of this interface.